app/script/user/UserRepository.js

```diff
@@ -0,0 +1,925 @@
+   * @private
+   */
+  _assign_connection(user_et) {
+    const connection_et = this.get_connection_by_user_id(user_et.id);
+    if (connection_et) {
+      user_et.connection(connection_et);
+    }
+    return connection_et;
+  }
+
+  /**
+   * Update the status of a connection.
+   * @private
+   * @param {z.entity.User} user_et - User to update connection with
+   * @param {string} status - Connection status
+   * @param {boolean} show_conversation - Show conversation on success
+   * @returns {Promise} Promise that resolves when the connection status was updated
+   */
+  _update_connection_status(user_et, status, show_conversation = false) {
+    return Promise.resolve()
+      .then(() => {
+        if (user_et.connection().status() === status) {
+          return;
+        }
+        return this.user_service.update_connection_status(user_et.id, status);
+      })
+      .then((response) => {
+        return this.user_connection(response, show_conversation);
+      })
+      .catch((error) => {
+        this.logger.error(`Connection status change to '${status}' for user '${user_et.id}' failed: ${error.message}`, error);
+
+        const custom_data = {
+          current_status: user_et.connection().status(),
+          failed_action: status,
+          server_error: error,
+        };
+
+        Raygun.send(new Error('Connection status change failed'), custom_data);
+      });
+  }
+
+  /**
+   * Convert a JSON event into an entity and get the matching conversation.
+   * @param {Object} event_json - JSON data of 'user.connection' event
+   * @param {boolean} show_conversation - Should the new conversation be opened?
+   * @returns {undefined} No return value
+   */
+  user_connection(event_json, show_conversation) {
+    if (event_json == null) {
+      return;
+    }
+    event_json = event_json.connection || event_json;
+
+    let connection_et = this.get_connection_by_user_id(event_json.to);
+    let previous_status = null;
+
+    if (connection_et != null) {
+      previous_status = connection_et.status();
+      this.connection_mapper.update_user_connection_from_json(connection_et, event_json);
+    } else {
+      connection_et = this.connection_mapper.map_user_connection_from_json(event_json);
+      this.update_user_connections([connection_et]);
+    }
+
+    if (connection_et) {
+      if ((previous_status === z.user.ConnectionStatus.SENT) && (connection_et.status() === z.user.ConnectionStatus.ACCEPTED)) {
+        this.update_user_by_id(connection_et.to);
+      }
+      this._send_user_connection_notification(connection_et, previous_status);
+      amplify.publish(z.event.WebApp.CONVERSATION.MAP_CONNECTION, connection_et, show_conversation);
+    }
+  }
+
+  /**
+   * Use a JSON event to update the matching user.
+   * @param {Object} event_json - JSON data
+   * @returns {z.entity.User} Updated user entity.
+   */
+  user_update(event_json) {
+    if (event_json.user.id === this.self().id) {
```

Comment:
I could do `user_update({user}) { ... }` but how should I document it then in JSDoc's `@param`?